source/agora/node/Node.d

```diff
@@ -248,6 +248,16 @@ public class Node : API
             this.network.sendTransaction(tx);
             this.ledger.tryNominateTXSet();
         }
+
+        if (this.enroll_man.needRevealPreimage(this.ledger.getBlockHeight()))
```

Comment:
This is not the right place to put this. Why is it being triggered when we receive a Transaction? There might not be enough transactions in the pool to start a nomination. It should be triggered when the ledger's state changes (new block added / externalize was called), and only if we're not catching up on outdated blocks.

Reply:
https://github.com/bpfkorea/agora/issues/582